core/tests/puppeteer-acceptance-tests/utilities/user/logged-in-user.ts

```diff
@@ -176,6 +176,8 @@ export class LoggedInUser extends BaseUser {
    * Navigates to the learner dashboard.
    */
   async navigateToLearnerDashboard(): Promise<void> {
+    // Reloading the page to ensure the latest state is reflected.
+    await this.page.reload();
```

Comment:
@HardikGoyal2003 This isn't really an acceptable solution, I think. If the test is instead navigating to learnerDashboardUrl by typing the same URL in again, could you please instead make the test do what a user would actually do in practice (i.e. use controls on the page to go to the right place)?

That said, I do agree that this issue (of re-typing the same URL) is one that we should try to prevent in our acceptance tests. Per https://pptr.dev/api/puppeteer.page.goto it seems that navigation to the same URL will succeed and return null ... maybe one thing we could do is to add logic in this.goto() to detect a null response and throw an error in such cases along the lines of "Attempting to navigate to the same URL that the user is already on by typing it in the URL bar; please use more natural user actions instead."? What do you think about this?

Also /cc @imchristie for her thoughts.

Reply:
Done!